addons/website/static/src/scss/website.scss

```diff
@@ -1012,3 +1012,35 @@ $ribbon-padding: 100px;
         bottom: 0;
     }
 }
+
+// Background shapes:
+@function str-replace($string, $search, $replace: '') {
+    $index: str-index($string, $search);
+
+    @if $index {
+        @return str-slice($string, 1, $index - 1) + $replace + str-replace(str-slice($string, $index + str-length($search)), $search, $replace);
+    }
+
+    @return $string;
+}
+
+.o_we_shape {
+    @each $module, $shapes in $bg-shapes {
+        @each $shape, $style in $shapes {
+            $url-params: '';
+            $colors: map-get($style, colors);
+            @each $i in $colors {
+                // %23 is the url-encoded form of '#' which doesn't work as is in urls.
+                $color: str-replace("#{map-get($o-color-palette, "o-color-#{$i}")}", '#', '%23');
+                $url-params: '#{$url-params}&c#{$i}=#{$color}';
+            }
+
+            // eg: o_website_shape_bg_1
+            &.o_#{$module}_#{$shape} {
+                background-image: url("/web_editor/shape/#{$module}/#{$shape}.svg?#{str-slice($url-params, 2)}");
+                background-position: map-get($style, position);
+                background-size: map-get($style, size);
+            }
+        }
+    }
+}
```

Comment:
I had in mind those classes were not necessary anymore... but maybe I got confused, we need to talk about this tomorrow

Reply:
Leaving a comment for posterity as we discussed this in person.

These classes allow non-customized shapes to update in real-time when changing the theme's color palette, which was a feature that FP really wanted.